Chatkit/Data Providers/TypingUsersProvider.swift

```diff
@@ -27,10 +27,13 @@ public class TypingUsersProvider {
         return Set(users)
     }
     
+    let currentUserIdentifier: String
```

Comment:
I still don't see that this at belonging in here when it's not used here

Reply:
OK, in that case should we instantiate the view model with an instance of `Chatkit` in order to be sure that the user passed to the view model is actually the one that is currently logged into the system?

Reply:
It look like we have 3 options here:

1. Leave it as it is right now and hide the `currentUser` from developers:
   - The public interface is simpler.
   - It is not possible to instantiate the view model with an incorrect instance of `User`.
   - Unfortunately, the identifier needs to be passed here somehow (internally).
2. Instantiate the view model with an instance of `Chatkit`:
   - The public interface is a bit more complex simpler.
   - It is still not possible to instantiate the view model with an incorrect instance of `User` (that's a good thing).
   - It **is** possible to instantiate the view model with an incorrect instance of `Chatkit`.
   - Initializer of the view model will have to be be optional (error handling would be optional as well).
3. Instantiate the view model with an instance of `User` or an identifier of the current user (`String`):
   - The public interface is a bit more complex simpler.
   - It **is** possible to instantiate the view model with an incorrect instance of `User` (that's not a good thing).
   - A bit cleaner implementation (internally).

Reply:
Hrm. I see what you mean. Personally, I'd definitely request it from the TypingIndicatorsViewModel constructor - I was even thinking to turn it in to a generic "filter user" parameter. I'm unsure about the correctness / ease of use tradeoff here.

There is a 4th option, which Zan suggested:
- create view models *from* the Chatkit object, like we do with providers.

I think this is definitely the most ergonomic, but it also makes it seem more "magic" to the customer, which would discourage them (or make a different way of implementing at least) their own view models in future.

Reply:
After the recent change all view models are now being initialized using factory methods.